### DISCUSSION TITLE
How to get continuous tracking during inference

### DISCUSSION THREAD

**Discussion by EBGU** [talmolab/sleap#741] [Help!]
Dear All,
First, thank you for your great software SLEAP!
However, I am currently facing a problem that I could not get a continuous tracking result. I trained a top-down model with a 3000-frame video with each frame labeled manually. After inference and tracking, I got 87 tracks for 2 animals. Because I am working with courtship behaviors, I need two tracks (one for female and one for male) to get features like male wing angle for behavior classification. What I got was a lot of fragments, and some frames only had one track on them (like the image I attached). Manual corrections are too expensive for a dozen of videos, but I really need intact tracks so that I could extract features to train a classifier. Could you help me with it?
Thank you very much!
![Picture2](https://user-images.githubusercontent.com/34360222/166679983-d35d4e8f-06cd-4d49-9363-9b0ba38d0c6c.jpg)

**Comment by talmo** [comment]
Hi @EBGU,

If you check out the [proofreading](https://sleap.ai/tutorials/proofreading.html) part of the tutorial, you'll see that you can set a number of different parameters related to tracking once you're on that step.

For this case, what you can do is set the **Target Number of Instances** to 2 and enable **Cull Instance Count**. I'd also recommend enabling **Connect Single Track Breaks**. These should all help cut down on the amount of switching you get, but let us know how it works for you after you get a chance to try it out and we'll go from there.

Cheers,

Talmo

**Comment by EBGU** [comment]
> 

Hi Talmo,

Thank you for your reply!
I have already done **setting the Target Number of Instances to 2 and enable Cull Instance Count** when I posted the issue. I also tried to enable **Connect Single Track Breaks** this time, but unfortunately, it didn't help. After enabling, I got 84 tracks instead of 87. Here is my configuration(the first attachment). Some frames still had only one instance, while some frames had instances without tracking information(the second and third attachment). My version is SLEAP 1.2.2.

Cheers,
Harold
![image](https://user-images.githubusercontent.com/34360222/166853268-09f39a57-ab58-4891-92c6-219c7434bcb3.png)
![Picture1](https://user-images.githubusercontent.com/34360222/166853600-c7b9bcaf-2714-48bc-a9a6-43ba772bb041.jpg)
![Picture2](https://user-images.githubusercontent.com/34360222/166853609-fdd12c60-9317-479b-b07e-8b0652fd5aae.jpg)

**Reply by EBGU** [reply]
I found the Kalman filter caused the problem.

**Reply by roomrys** [reply]
@EBGU Thanks for coming back with your find. I have a few questions just for clarification: Did you forgo using the Kalman Filter entirely? Or did you end up just tweaking the settings of the Kalman filter (i.e. not enough initial frames, using extremities as the tracking nodes, etc.)?

**Comment by talmo** [comment]
Hi @EBGU,

Sorry for the delay! We've been a bit swamped with support requests lately. This is more of a usage problem than a bug, so I'm going to move this to our discussions page. We'll follow up over there.

Talmo
